docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,709 @@
+
+1. [What is a Turbomodule, In Short](#what-is-a-turbomodule-in-short)
+1. [Create the Native Module](#create-the-native-module)
+   1. [Creating the Structure](#creating-the-structure)
+   1. [Create a Testing Environment](#create-a-testing-environment)
+   1. [Define the JS Shared Code](#define-the-js-shared-code)
+   1. [Implement the iOS Code](#implement-the-ios-code)
+   1. [Implement the Android Code](#implement-the-android-code)
+   1. [Test the Native Module](#test-the-native-module)
+1. [Create the Turbomodule](#create-the-turbomodule)
+   1. [Create the New Architecture Testing Environment](#create-the-new-architecture-testing-environment)
+   1. [Adding the JS Spec](#adding-the-js-spec)
+   1. [Generate the Code](#generate-the-code)
+   1. [Update the iOS Code](#update-the-ios-code)
+   1. [Update the Android Code](#update-the-android-code)
+   1. [Test the Turbomodule](#test-the-turbomodule)
+1. [Summary](#summary)
+
+## What Is a Turbomodule, In Short.
+
+Turbomodules are the evolution of Native Modules. The main benefit of using a Turbomodule is that the communication between the native layer and the Javascript layer happens using shared memory instead of the bridge. This allows for better performances and synchronous calls between the native code and Javascript, while the bidge forced us to work with asynchronous communication where all the messages were serialized using a JSON format.
+
+The communication is handled by a C++ layer and this code is generated automatically from the JS code. Native developers don't have to write C++, if not for some very small bits. The C++ code is shared across all the supported platforms (iOS, Android, ...): we don't have different implementation for each of them. This improve maintainability and uniformity across platform.
+
+<!-- To learn more about Turbomodule, check this [deep dive](). -->
+
+## Create the Native Module
+
+This section and the following are a practical guides to create a module that is backward compatible. We are not diving deep into what can be done with Native Modules. For more informations on that topic, have a look [here](native-modules-intro).
+
+If you are interested in creating a Turbomodule, you can skip this part and directly go [here](#create-the-turbomodule). Consider that most of the code of a Native Module is needed for a Turbomodule.
+
+### Creating the Structure
+
+We have two alternatives to create the basic structure for our module.
+
+1. Using the React Native CLI
+2. Manually.
+
+#### React Native CLI
+
+To create a turbomodule with the React Native CLI, run the following command:
+
+```sh
+npx create-react-native-library <name-of-the-library>
+```
+
+Follow the questions the CLI asks and, once the process is done, it will create a folder structure completed with an example app, typescript configuration and much more. The sample structure is shown in the following image:
+
+<center><img src="docs/assets/Turbomodules/NpxLibFolderStructure.png" /></center>
```

Comment:
```suggestion
Follow the questions the CLI asks and, once the process is done, it will create a folder structure completed with an example app, typescript configuration and much more.

<figure>
  <img src="docs/assets/Turbomodules/NpxLibFolderStructure.png" />
  <figcaption>The sample directory structure.</figcaption>
</figure>
```

If the image needs a description, we can use `figure` and `figcaption` tags, which also is more semantic correct.